execution/executor/src/speculation_cache/mod.rs

```diff
@@ -189,6 +189,8 @@ impl SpeculationCache {
             );
             id
         };
+        // prune the old root tree
+        self.committed_trees.state_tree().prune();
```

Comment:
I understand you think postponing the pruning of the newly committed block is beneficial to performance. But I'd rather make a straight forward fix in this by removing the "rev().skip(1)" part, and add this hack in a separate PR and document the performance difference (via cluster test result, for example.)